tools/extract_cmake_functions.py

```diff
@@ -56,6 +56,49 @@
 
 all_functions = {}
 
+# Supported commands:
+# \brief\ <brief description, which should be included in the main description>
+# \brief_nodesc\ <brief description, which should be excluded from the main description>
+# \param\ <parameter_name> <parameter description>
```

Comment:
IMHO it might be nicer if these were `\brief`, `\param`, etc. so that they're share the same "syntax" as doxygen commands, rather than requiring them to also have a `\` at the end.
You could parse this by changing:
```python
        line = line.strip()
        if line.startswith('\\'):
            command = line.split('\\')[1]
```
to something like
```python
        line = line.strip()
        m = re.match('\\(\w+)', line)
        if m:
            command = m.group(1)
```
?

Reply:
I was going for similar commands but different syntax to doxygen (hence the second slash), because the parsing is entirely different and will not handle the vast majority of doxygen stuff, so I wanted to prevent copy/pasting but still have similar commands

Reply:
Yeah, the script should throw an error if the command isn't known, which would happen if it didn't have the second backslash
```python
logger.error("{}:{} has unknown command: {}".format(group, name, command))
```